### ISSUE TITLE
About LuBu and DiaoChan

### ISSUE THREAD

**Issue by Billmvp73** [ve280/ve280#30]
I think the implementation of LuBu and DiaoChan should be different from other heroes. Their skills change the effect of an existing card and a player's behavior. So I'm confused about how to write their `const Card * castCard(const Card *card)`.  Maybe I can create new derived classes of `class card` to fulfill their functions, like `class lubustrike`. Anyone has better ideas?

**Comment by zianke** [comment]
Yes I agree that you can create a new derived Card class which implements a different `Card::takeEffect()` function. 
Interestingly in my solution the class is named exactly as `LuBuStrike` : )
